File: src/Trinity.TSL/Trinity.TSL.CodeGen.Legacy/Metagen/Enum.cs.cs
using System.Text;
using System.Collections.Generic;

namespace Trinity.TSL.CodeTemplates
{
    internal partial class SourceFiles
    {
        internal static string 
Enum(
NTSL node)
        {
            StringBuilder source = new StringBuilder();
            
source.Append(@"using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Trinity;
using Trinity.TSL;
namespace ");
source.Append(Codegen.GetString(Trinity::Codegen::GetNamespace()));
source.Append(@"
{
    ");
for (int iterator_1 = 0; iterator_1 < (node->enumList).Count;++iterator_1)
{
source.Append(@"
    /// <summary>
    /// Represents the enum type ");
source.Append(Codegen.GetString((node->enumList)[iterator_1].name));
source.Append(@" defined in the TSL.
    /// </summary>
    public enum ");
source.Append(Codegen.GetString((node->enumList)[iterator_1].name));
source.Append(@" : byte
    {
        ");
for (int iterator_2 = 0; iterator_2 < ((node->enumList)[iterator_1].enumEntryList).Count;++iterator_2)
{
source.Append(Codegen.GetString(((node->enumList)[iterator_1].enumEntryList)[iterator_2].name));
source.Append(@" = ");
source.Append(Codegen.GetString(((node->enumList)[iterator_1].enumEntryList)[iterator_2].value));
if (iterator_2 < ((node->enumList)[iterator_1].enumEntryList).Count - 1)
source.Append(",");
}
source.Append(@"
    }
    ");
}
source.Append(@"
}
");

            return source.ToString();
        }
    }
}

File: src/Trinity.TSL/Trinity.TSL.CodeGen/Enum.cs.cpp
#include "common.h"
#include <string>
#include <SyntaxNode.h>

using std::string;

namespace Trinity
{
    namespace Codegen
    {
        string* 
Enum(
NTSL* node)
        {
            string* source = new string();
            
source->append(R"::(using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Trinity;
using Trinity.TSL;
namespace )::");
source->append(Codegen::GetString(Trinity::Codegen::GetNamespace()));
source->append(R"::(
{
    )::");
for (size_t iterator_1 = 0; iterator_1 < (node->enumList)->size();++iterator_1)
{
source->append(R"::(
    /// <summary>
    /// Represents the enum type )::");
source->append(Codegen::GetString((*(node->enumList))[iterator_1]->name));
source->append(R"::( defined in the TSL.
    /// </summary>
    public enum )::");
source->append(Codegen::GetString((*(node->enumList))[iterator_1]->name));
source->append(R"::( : byte
    {
        )::");
for (size_t iterator_2 = 0; iterator_2 < ((*(node->enumList))[iterator_1]->enumEntryList)->size();++iterator_2)
{
source->append(Codegen::GetString((*((*(node->enumList))[iterator_1]->enumEntryList))[iterator_2]->name));
source->append(R"::( = )::");
source->append(Codegen::GetString((*((*(node->enumList))[iterator_1]->enumEntryList))[iterator_2]->value));
if (iterator_2 < ((*(node->enumList))[iterator_1]->enumEntryList)->size() - 1)
source->append(",");
}
source->append(R"::(
    }
    )::");
}
source->append(R"::(
}
)::");

            return source;
        }
    }
}
